Contents/Code/__init__.py

```diff
@@ -72,51 +76,127 @@ class ShokoCommonAgent:
     def Search(self, results, media, lang, manual, movie):
         name = ( media.title if movie else media.show )
 
-        # http://127.0.0.1:8111/api/serie/search?query=Clannad&level=1&apikey=d422dfd2-bdc3-4219-b3bb-08b85aa65579
-
         if movie:
             if media.filename:
-                filename = os.path.basename(urllib.unquote(media.filename))
-
-                episode_data = HttpReq("api/ep/getbyfilename?filename=%s" % (urllib.quote(filename.encode('utf8'))))
-                movie_data = HttpReq("api/serie/fromep?id=%s" % (episode_data['id']))
-
-                score = 100 if movie_data['name'] == name else 85  # TODO: Improve this to respect synonyms./
-                title = movie_data['name'] + ' - ' + episode_data['name']
-                meta = MetadataSearchResult('%s' % (episode_data['id']), title, try_get(episode_data, 'year', None), score, lang)
+                filename = urllib.unquote(media.filename)
+                filename = os.path.join(os.path.split(os.path.dirname(filename))[-1], os.path.basename(filename)) # Parent folder + file name
+
+                Log('Searching movie %s - %s' % (name, filename))
+
+                # Get file data using filename
+                # http://127.0.0.1:8111/api/v3/File/PathEndsWith/%5Bjoseole99%5D%20Clannad%20-%2001%20(1280x720%20Blu-ray%20H264)%20%5B8E128DF5%5D.mkv
+                file_data = HttpReq('api/v3/File/PathEndsWith/%s' % (urllib.quote(filename)))
+
+                # Take the first file. As we are searching with both parent folder and filename, there should be only one result.
+                if len(file_data) > 1:
+                    Log('File search has more than 1 result. HOW DID YOU DO IT?')
+                file_data = file_data[0]
+
+                # Get series data
+                series_id = file_data['SeriesIDs'][0]['SeriesID']['ID'] # Taking the first matching anime. Not supporting multi-anime linked files for now. eg. Those two Toradora/One Piece episodes
+                series_data = {}
+                series_data['shoko'] = HttpReq('api/v3/Series/%s' % series_id) # http://127.0.0.1:8111/api/v3/Series/24
+                series_data['anidb'] = HttpReq('api/v3/Series/%s/AniDB' % series_id) # http://127.0.0.1:8111/api/v3/Series/24/AniDB
+
+                # Get episode data
+                ep_id = file_data['SeriesIDs'][0]['EpisodeIDs'][0]['ID'] # Taking the first
```

Comment:
Might be worth making this more error resistant, since what if no series or episode IDs are identified?

Reply:
We are ignoring unrecognized files in the scanner. Any file which doesn't have a series ID won't be here, and if a file has a series ID linked to it, that cannot happen unless it also has an episode ID on it since it is "file -> episode -> series"
It doesn't require it IMO but added it anyway on line 96 because... edge cases.